flutter-studio/src/io/flutter/module/FlutterModuleImporter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.module;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+/**
+ * In an Android project, edit the settings.gradle and the app/build.gradle files as described in the add2app spec.
+ */
+public class FlutterModuleImporter {
+  private final FlutterModuleModel myModel;
+  private boolean hasFinishedEditing = false;
+  private StringWriter writer;
+  private String myRelativePath;
+  private String myNewline;
+
+  public FlutterModuleImporter(FlutterModuleModel model) {
+    myModel = model;
+    myNewline = System.lineSeparator();
+  }
+
+  public void importModule() {
+    String location = myModel.projectLocation().get();
+    VirtualFile moduleRoot = VfsUtil.findFileByIoFile(new File(location), true);
+    assert (moduleRoot != null);
+    assert (myModel.project().get().isPresent());
+    Project androidProject = myModel.project().get().get();
+    VirtualFile projectRoot = androidProject.getBaseDir();
+    myRelativePath = io.flutter.utils.System.findRelativePath(projectRoot.getParent(), moduleRoot, File.separatorChar);
+    if (myRelativePath == null) {
+      showHowToEditDialog();
+      return;
+    }
+
+    myRelativePath = Paths.get(myRelativePath, ".android", "include_flutter.groovy").normalize().toString();
+    VirtualFile settingsFile = projectRoot.findChild("settings.gradle");
+    if (settingsFile == null) {
+      showHowToEditDialog();
```

Comment:
👍, I like that if things go off the rails, we give the user some manual step forward